integration-test/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/ProcessUtils.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.k8s.integrationtest
+
+import java.util.concurrent.TimeUnit
+
+import scala.collection.mutable.ArrayBuffer
+import scala.io.Source
+
+object ProcessUtils extends Logging {
+  /**
+   * executeProcess is used to run a command and return the output if it
+   * completes within timeout seconds.
+   */
+  def executeProcess(fullCommand: Array[String], timeout: Long): Seq[String] = {
+    val pb = new ProcessBuilder().command(fullCommand: _*)
```

Comment:
Why not launch a pod to kick off the spark-submit command?

Reply:
People launch `spark-submit` from outside their k8s clusters. I think launching spark-submit itself inside a pod would deviate from how people use spark on k8s. Besides, I don't even know running spark-submit inside a pod works :-)

Reply:
I believe decoupling the spark-submit portion from the integration tests is better to assume that issues with host machine don't cause the integration tests to fail when they shouldn't. (we have enough assumptions already with Python and R). Thoughts? I can send up a PR to add launching from a pod if we believe it to be better than using ProcessBuilder, which I am trying to avoid.

Reply:
Launching from a pod would add a second layer of indirection when trying to debug issues with the spark-submit runtime itself. Making assumptions about the host machine should be fine, isn't that what the Spark R and Python unit tests do?

Reply:
Also it would complicate auth issues in the future when we add back custom client credentials etc. See [DriverKubernetesCredentialsStep](https://github.com/apache-spark-on-k8s/spark/blob/branch-2.2-kubernetes/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/submitsteps/DriverKubernetesCredentialsStep.scala]).